### PR TITLE
Python test code

### PR DESCRIPTION
**Pull request by koivunej**
Starts the story for `py/` which will have the external process call functionality etc. test utils.

### CODE REVIEW

**Review by Mirko-von-Leipzig — APPROVED** [review]
Looks good, just one suggestion. And then I would add a usage example and maybe add it to a tools / utils section in the readme for discoverability?

File: py/src/generate_test_storage_tree.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# # generate_test_tree.py
+#
+# read stdin for lines of "key value", after closing stdin will report a root
+# hash on stdout for this per-contract storage merkle tree. nodes will be
+# dumped on stderr.
+#
+# keys and values are either:
+#
+# - hex for big endian integers (whatever accepted by bytes.fromhex)
```
[the remark this answers is not in the document]
Comment (koivunej) [reply py/src/generate_test_storage_tree.py:9]: Oh fumbled this out. I'll add the echo example.

File: py/src/generate_test_storage_tree.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# # generate_test_tree.py
```
Comment (koivunej) [inline py/src/generate_test_storage_tree.py:1]:
```suggestion
# # generate_test_tree.py
#
# usage: echo "1 2" | python src/generate_test_storage_tree.py
```

**Comment by koivunej** [comment]
I was thinking this would be better linked from under the crates/pathfinder/src/merkle_tree.rs as for it's usefulness/usecase, produce test trees.

**Comment by Mirko-von-Leipzig** [comment]
> > maybe add it to a tools / utils section in the readme for discoverability?
> 
> I was thinking this would be better linked from under the crates/pathfinder/src/merkle_tree.rs as for it's usefulness/usecase, produce the correct test trees.

I'll definitely link it there. But I'm thinking more for people exploring the code base. Can be confusing to just fall into a folder and have no idea what that's for.

**Comment by Mirko-von-Leipzig** [comment]
Should we also add `.venv` to `.gitignore`?

**Comment by koivunej** [comment]
> Should we also add `.venv` to `.gitignore`?

Isn't it being ignored for you? It is already in `py/.gitignore`, along with the `__pycache__` (generation of this could be disabled in other ways but ... I thinks good enough like this).

**Comment by koivunej** [comment]
Oki after a few fumblings I think this is looking good to merge, merging soonish.

**Review by CHr15F0x — APPROVED** [review]
LGTM! I don't think I have anything to add unless I play a bit with the test myself :wink: 
Sorry for a late review :sweat_smile: .

**Pull request merged**